### PR TITLE
Refactoring selections and TableItems component 

### PR DESCRIPTION
**Pull request by siemiatj**
Related to #6873 #6854 

Apart of fixing running processes from the actions it also removes some unnecessary code we've had in table rows. All changes are now handled via websocket events so it's not needed anymore.

### CODE REVIEW

File: frontend/src/utils/tableHelpers.js [hunk 1]
```diff
@@ -121,6 +121,24 @@ export function getSizeClass(col) {
   }
 }
```
Comment (petrican) [inline frontend/src/utils/tableHelpers.js:123]: Pls add Doc this + param. Thanks

**Comment by petrican** [comment]
https://www.loom.com/share/b57ca84903d44950b8684f81be0d6c73
Video with issues that needs to be fixed before merging

**Comment by petrican** [comment]
Another issue that I saw during dev testing. https://www.loom.com/share/ac210d52623b4409a44cb3707a0153d2
The field does not update instantaneously. Pls see the video. That is not a pleasant UX experience

File: frontend/src/utils/documentListHelper.js [hunk 2]
```diff
@@ -100,6 +98,24 @@ const DLmapStateToProps = (state, props) => {
     viewId = null;
   }
 
+  let childTableId = null;
+  const childSelector = getSelection();
+  if (props.includedView && props.includedView.windowType) {
```
Comment (petrican) [inline frontend/src/utils/documentListHelper.js:103]: shrink the code by using destructuring  { includedView } pls

File: frontend/src/utils/documentListHelper.js [hunk 3]
```diff
@@ -100,6 +98,24 @@ const DLmapStateToProps = (state, props) => {
...
+    childTableId = getTableId({
+      windowId: props.includedView.windowType,
+      viewId: props.includedView.viewId,
+    });
+  }
+
+  let parentTableId = null;
+  const parentSelector = getSelection();
+  if (props.parentWindowType) {
```
Comment (petrican) [inline frontend/src/utils/documentListHelper.js:112]: const { parentWindowType, parentDefaultViewId } = props;

File: frontend/src/components/app/RawModal.js [hunk 4]
```diff
@@ -409,38 +405,46 @@ ModalButton.propTypes = {
...
-  masterDocumentList: PropTypes.any,
-  modalTitle: PropTypes.oneOfType([PropTypes.string, PropTypes.node]),
-  modalDescription: PropTypes.oneOfType([PropTypes.string, PropTypes.node]),
   modalVisible: PropTypes.bool,
   rawModal: PropTypes.object.isRequired,
   requests: PropTypes.object.isRequired,
   success: PropTypes.bool.isRequired,
+  indicator: PropTypes.string.isRequired,
+  modalTitle: PropTypes.oneOfType([PropTypes.string, PropTypes.node]),
+  modalDescription: PropTypes.oneOfType([PropTypes.string, PropTypes.node]),
```
[the remark this answers is not in the document]
Comment (siemiatj) [reply frontend/src/components/app/RawModal.js:429]: I wish, but I think for some reason sometimes we're sending a node. But this is not the focus of this PR so I won't dive into that. If you have spare cycles you can try to figure that out.

File: frontend/src/components/header/SubHeader.js [hunk 5]
```diff
@@ -503,26 +500,23 @@ class SubHeader extends Component {
...
-        activeTabSelected={activeTab && selected ? selected : []}
       />
     );
   };
 
-  /**
-   * @method render
-   * @summary ToDo: Describe the method.
-   */
+  setRef = (ref) => {
```
[the remark this answers is not in the document]
Comment (siemiatj) [reply frontend/src/components/header/SubHeader.js:509]: For some reason I thought our linter settings forbid that.

File: frontend/src/components/app/MasterWindow.js [hunk 6]
```diff
@@ -295,6 +295,8 @@ export default class MasterWindow extends PureComponent {
       introHints,
     } = this.state;
     const { docActionElement, documentSummaryElement } = master.layout;
+
+    // TODO: Do we need to have docId and dataId ?
```
[the remark this answers is not in the document]
Comment (siemiatj) [reply frontend/src/components/app/MasterWindow.js:299]: Maybe. Still this is not the scope of this effort, so just leaving it here.

File: frontend/src/components/table/TableItem.js [hunk 7]
```diff
@@ -360,19 +379,22 @@ class TableItem extends PureComponent {
       isGerman,
       isSelected,
       focusOnFieldName,
-      activeLocale,
+      /*
```
[the remark this answers is not in the document]
Comment (siemiatj) [reply frontend/src/components/table/TableItem.js:390]: ? What's this comment all about ?

**Comment by siemiatj** [comment]
> Another issue that I saw during dev testing. https://www.loom.com/share/ac210d52623b4409a44cb3707a0153d2
> The field does not update instantaneously. Pls see the video. That is not a pleasant UX experience

Yeah, this is caused by the fact we don't update anything locally anymore but react to ws events from the server. This would work better if backend wasn't so slow, but for now I have to think of a way of bringing this behavior back without complicating things too much.

File: frontend/src/components/table/TableItem.js (hunk 7 above)
[the remark this answers is not in the document]
Comment (petrican) [reply frontend/src/components/table/TableItem.js:390]: is about removing or not `onItemChange`

**Comment by petrican** [comment]
> > Another issue that I saw during dev testing. https://www.loom.com/share/ac210d52623b4409a44cb3707a0153d2
> > The field does not update instantaneously. Pls see the video. That is not a pleasant UX experience
> 
> Yeah, this is caused by the fact we don't update anything locally anymore but react to ws events from the server. This would work better if backend wasn't so slow, but for now I have to think of a way of bringing this behavior back without complicating things too much.

Possible solution would be to keep in state the value that you are editing (at widget level) and sync it with the result of the patch (getDerivedStateFromProps). Patch should be called with a debounce when user stops typing.

**Comment by petrican** [comment]
Build was failing for some reason. Triggered a rebuild. Waiting for it to pass.

**Pull request merged**